### PR TITLE
Adding in lesson09 Unfinished

### PR DESCRIPTION
**Pull request by cholso**
Unfortunately this is quite unfinished ): Planning on finishing it later regardless of tardiness to learn OO.

### CODE REVIEW

**Comment by charlesrobison** [comment]
@cholso - with the 2 week extension because of the instructors being out your end date is July 21st.  You still have time to finish - do you want to close this PR and submit when complete?

**Comment by cholso** [comment]
Thank you for the heads-up! Unfortunately I am away on a trip right now and
have limited access to computers. I am fine submitting the pull request now
and completing the assignment for my own learning later. Hope you have a
wonderful day!

Best,
@cholso

**Comment by natasha-aleksandrova** [comment]
@charles please go ahead and submit the grade, merging this.

**Comment by natasha-aleksandrova** [comment]
@charles please go ahead and submit the grade, merging this.

File: students/chelsea_nayan/lesson09/mailroom_oo.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    # Add a donation to a donor
+    def add_donation(self, donation):
+        self.donations.append(donation)
+
+    #donor_names, total_given, num_gifts, average_gift = [], [], [], []
+    def donation_nums(self):
+        num_gifts = len(self.donations)
+        total_given = sum(self.donations)
+        average_gift = total_given / num_gifts
+        donation_nums_for_report = [total_given, num_gifts, average_gift]
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/mailroom_oo.py]: This variable isn't being used

File: students/chelsea_nayan/lesson09/mailroom_oo.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return donor_info
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/mailroom_oo.py]: donor_info isn't defined

File: students/chelsea_nayan/lesson09/mailroom_oo.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        self.donor_list = []
+
+    # Return a list of donor names
+    def donor_names(self):
+        self.names[]
+        return [donor[0] for donor in self.donor_list]
+
+    # Add donor information for report
+    def add_donor(self, donor):
+        self.donor_list.append(donor.name, donor_info)
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/mailroom_oo.py]: donor_info here as well

File: students/chelsea_nayan/lesson09/mailroom_oo.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+
+# Hold donor objects and methods, add new donors, search for a donor, generate a report about a donor
+class DonorCollection():
+    def __init__(self):
+        self.donor_list = []
+
+    # Return a list of donor names
+    def donor_names(self):
+        self.names[]
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/mailroom_oo.py]: Syntax error - do you mean self.names = []?

File: students/chelsea_nayan/lesson09/mailroom_oo.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        for donors in self.donor_list:
+            donor_names += donors.name + '\n'
+        return donor_names
+
+    # Send the intial donor dictionary to DonorCollection
+    @staticmethod
+    def initial_donors(donor_info):
+        donors = DonorCollection()
+        for name, amounts in donor_info.items():
+        donor = Donor(name, amounts)
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/mailroom_oo.py]: Indentation needed here

File: students/chelsea_nayan/lesson09/cli_main.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                2: report,
+                3: send,
+                4: quit
+    }
+
+    # A loop to prompt the user for a response of one of four options
+    while True:
+        response = input('''
+    Choose one of four actions:
+    [1] Send a Thank You
```
Comment (charlesrobison) [inline students/chelsea_nayan/lesson09/cli_main.py]: Getting an error when trying to add a donor - throws the TypeError from Donor() class in mailroom_oo.py

**Pull request merged**